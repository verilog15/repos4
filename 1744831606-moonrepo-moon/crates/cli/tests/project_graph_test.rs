use moon_test_utils::{
    assert_snapshot, create_sandbox_with_config, get_project_graph_aliases_fixture_configs,
    get_projects_fixture_configs,
};

#[test]
fn no_projects() {
    let sandbox = create_sandbox_with_config("base", None, None, None);

    let assert = sandbox.run_moon(|cmd| {
        cmd.arg("project-graph").arg("--dot");
    });

    assert_snapshot!(assert.output());
}

#[test]
fn many_projects() {
    let (workspace_config, toolchain_config, tasks_config) = get_projects_fixture_configs();

    let sandbox = create_sandbox_with_config(
        "projects",
        Some(workspace_config),
        Some(toolchain_config),
        Some(tasks_config),
    );

    let assert = sandbox.run_moon(|cmd| {
        cmd.arg("project-graph").arg("--dot");
    });

    assert_snapshot!(assert.output());
}

#[test]
fn single_project_with_dependencies() {
    let (workspace_config, toolchain_config, tasks_config) = get_projects_fixture_configs();

    let sandbox = create_sandbox_with_config(
        "projects",
        Some(workspace_config),
        Some(toolchain_config),
        Some(tasks_config),
    );

    let assert = sandbox.run_moon(|cmd| {
        cmd.arg("project-graph").arg("foo").arg("--dot");
    });

    assert_snapshot!(assert.output());
}

#[test]
fn single_project_with_dependents() {
    let (workspace_config, toolchain_config, tasks_config) = get_projects_fixture_configs();

    let sandbox = create_sandbox_with_config(
        "projects",
        Some(workspace_config),
        Some(toolchain_config),
        Some(tasks_config),
    );

    let assert = sandbox.run_moon(|cmd| {
        cmd.arg("project-graph")
            .arg("bar")
            .arg("--dot")
            .arg("--dependents");
    });

    assert_snapshot!(assert.output());
}

#[test]
fn single_project_no_dependencies() {
    let (workspace_config, toolchain_config, tasks_config) = get_projects_fixture_configs();

    let sandbox = create_sandbox_with_config(
        "projects",
        Some(workspace_config),
        Some(toolchain_config),
        Some(tasks_config),
    );

    let assert = sandbox.run_moon(|cmd| {
        cmd.arg("project-graph").arg("baz").arg("--dot");
    });

    assert_snapshot!(assert.output());
}

#[test]
fn outputs_json() {
    let (workspace_config, toolchain_config, tasks_config) = get_projects_fixture_configs();

    let sandbox = create_sandbox_with_config(
        "projects",
        Some(workspace_config),
        Some(toolchain_config),
        Some(tasks_config),
    );

    let assert = sandbox.run_moon(|cmd| {
        cmd.arg("project-graph").arg("foo").arg("--json");
    });

    assert_ne!(assert.output(), "{}");
}

mod aliases {
    use super::*;

    #[test]
    fn uses_ids_in_graph() {
        let (workspace_config, toolchain_config, tasks_config) =
            get_project_graph_aliases_fixture_configs();

        let sandbox = create_sandbox_with_config(
            "project-graph/aliases",
            Some(workspace_config),
            Some(toolchain_config),
            Some(tasks_config),
        );

        let assert = sandbox.run_moon(|cmd| {
            cmd.arg("project-graph").arg("--dot");
        });

        assert_snapshot!(assert.output());
    }

    #[test]
    fn can_focus_using_an_alias() {
        let (workspace_config, toolchain_config, tasks_config) =
            get_project_graph_aliases_fixture_configs();

        let sandbox = create_sandbox_with_config(
            "project-graph/aliases",
            Some(workspace_config),
            Some(toolchain_config),
            Some(tasks_config),
        );

        let assert = sandbox.run_moon(|cmd| {
            cmd.arg("project-graph").arg("@scope/pkg-foo").arg("--dot");
        });

        assert_snapshot!(assert.output());
    }

    #[test]
    fn resolves_aliases_in_depends_on() {
        let (workspace_config, toolchain_config, tasks_config) =
            get_project_graph_aliases_fixture_configs();

        let sandbox = create_sandbox_with_config(
            "project-graph/aliases",
            Some(workspace_config),
            Some(toolchain_config),
            Some(tasks_config),
        );

        let assert = sandbox.run_moon(|cmd| {
            cmd.arg("project-graph").arg("noLang").arg("--dot");
        });

        assert_snapshot!(assert.output());
    }
}
