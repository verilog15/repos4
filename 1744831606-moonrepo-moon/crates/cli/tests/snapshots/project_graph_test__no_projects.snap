---
source: crates/cli/tests/project_graph_test.rs
assertion_line: 14
expression: assert.output()
---
digraph {
}



